gdk4/src/key_event.rs

```diff
@@ -0,0 +1,43 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{keys::Key, KeyEvent, KeyMatch, ModifierType};
+use glib::translate::*;
+use std::mem;
+
+impl KeyEvent {
+    #[doc(alias = "gdk_key_event_get_keyval")]
+    pub fn get_keyval(&self) -> Key {
+        unsafe { ffi::gdk_key_event_get_keyval(self.to_glib_none().0).into() }
+    }
+
+    #[doc(alias = "gdk_key_event_get_match")]
+    pub fn get_match(&self) -> Option<(Key, ModifierType)> {
+        unsafe {
+            let mut keyval = mem::MaybeUninit::uninit();
+            let mut modifiers = mem::MaybeUninit::uninit();
+            let ret = from_glib(ffi::gdk_key_event_get_match(
+                self.to_glib_none().0,
+                keyval.as_mut_ptr(),
+                modifiers.as_mut_ptr(),
+            ));
+            let keyval: Key = keyval.assume_init().into();
+            let modifiers = modifiers.assume_init();
```

Comment:
It's UB to call `assume_init()` on uninitialized data, which would be the case if `ret == false`. You first need to check `ret`.

Reply:
Same problem in other places FWIW

Reply:
That was part of the original code in key_even btw

Reply:
That's very bad :)